File: delivery-java-project/api/src/main/java/org/delivery/api/common/error/UserError.java
package org.delivery.api.common.error;

import org.springframework.http.HttpStatus;

import lombok.AllArgsConstructor;
import lombok.Getter;
/**
 * User의 경우 1000번대 에러 사용
 */

@AllArgsConstructor
// Getter를 달아줌으로써 ErrorCodeInterfcace에서 구현해야할 것들을 구현 안해줘도 된다.
@Getter
public enum UserError implements ErrorCodeInterface {
  USER_NOT_FOUND(400, 1404, "사용자를 찾을 수 없음"),
  ;
  
  /// HTTP 상태 코드
  private final Integer httpStatusCode;
  /// 서비스 에러 코드(개발자 정의)
  private final Integer errorCode; 
  private final String description;
}

File: delivery-java-project/common/src/main/kotlin/org/delivery/common/error/UserError.kt
package org.delivery.common.error

import lombok.Getter

/**
 * User의 경우 1000번대 에러 사용
 */
enum class UserError (
	private val httpStatusCode: Int,
	private val errorCode: Int,
	private val description: String
): ErrorCodeInterface {
  USER_NOT_FOUND(400, 1404, "사용자를 찾을 수 없음"),
  ;
	
	override fun getHttpStatusCode(): Int {
		return this.httpStatusCode
	}
	
	override fun getErrorCode(): Int {
		return this.errorCode
	}
	
	override fun getDescription(): String {
		return this.description
	}
}